Turkey/ProcessExtensions.cs

```diff
@@ -32,53 +32,20 @@ public static Task WaitForExitAsync(this Process process, CancellationToken toke
             process.ErrorDataReceived += errorHandler;
             process.BeginErrorReadLine();
 
-            var tcs = new TaskCompletionSource<bool>();
-            int completionOwner = 0;
-            CancellationTokenRegistration ctr = default;
-            ctr = token.Register(() => CompleteTask(cancel: true));
-            process.Exited += delegate { CompleteTask(cancel: false); };
-            if (process.HasExited) // In case the Process exited before we've added the handler.
+            try
             {
-                CompleteTask(cancel: false);
+                await process.WaitForExitAsync(token);
             }
-            return tcs.Task;
-
-            void CompleteTask(bool cancel)
+            catch (OperationCanceledException ex)
             {
-                // Only one caller can complete the task.
-                if (Interlocked.CompareExchange(ref completionOwner, 1, 0) == 1)
-                {
-                    return;
-                }
-
-                if (cancel)
-                {
-                    try
-                    {
-                        process.Kill();
-                    }
-                    catch // Before .NET 3, Kill can throw if the process has terminated in the meanwhile.
-                    { }
-
-                }
-                else
+                try
                 {
-                    process.WaitForExit();  // Wait until we've received all output.
-                    ctr.Dispose();          // Don't root objects via the CancellationToken.
+                    process.Kill(entireProcessTree: true);
                 }
+                catch
+                { }
```

Comment:
Should we log something here?

Reply:
This kill is best effort.
I'm not sure if it can actually throw. I've added the catch just in case.